dlib/entropy_decoder_model/entropy_decoder_model_kernel_5.h

```diff
@@ -244,14 +244,44 @@ namespace dlib
 
         inline void clear_exclusions (
         );
+
+        /*!
+            Helper for making sure parsing is not stuck -
+            problematic input may appear to hang the image parser
+            and during fuzzing, one wants to cut out early, to get
+            the fuzzing speed up. In production, one might want to
+            set a finite but high value for the limit.
+
+            Instead of while(true){...} use while(!checker()) {...}
+            and each time the counter is incremented, until the point it
+            the counter exceeds some threshold.
+        !*/
+
+        struct HangChecker {
```

Comment:
I would rather keep code like this out of the main dlib repo, or potentially in a branch.  But not on master. This includes any `#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION` type of thing.  That stuff should be in a branch or somewhere outside maser, and when errors are surfaced, fixed on master.

Reply:
I understand, but that makes the fuzzing prone to bit rot.
The macro is the defacto standard, see https://llvm.org/docs/LibFuzzer.html#fuzzer-friendly-build-mode

Here's an example from fmt: https://github.com/fmtlib/fmt/blob/6a497e1d061993cce54c2d71506a90155a3725e6/include/fmt/format.h#L599

Here are some recommendations from oss-fuzz:
https://github.com/google/oss-fuzz/blob/master/docs/ideal_integration.md#tldr

Reply:
I hear you.  But you are talking about parts of dlib that have not changed in many years, and are not likely to change. I doubt such bit rot will be a problem, and if this fuzzing ends up being used long term rather than being eventually abandoned and such problems do arise they can be handled then, maybe by merging into main.